tests/gptq/test_quantization.py

```diff
@@ -126,14 +128,89 @@ def test_serialization(self):
+    EXPECTED_OUTPUTS.add("Hello my name is jay and i am a student at university.")
+    EXPECTED_OUTPUTS.add("Hello my name is John, I am a student in the University of")
+
+
+class GPTQTestActOrder(GPTQTest):
+    EXPECTED_OUTPUTS = set()
+    EXPECTED_OUTPUTS.add("Hello my name is jay and i am a student at university.")
+    EXPECTED_OUTPUTS.add("Hello my name is jessie and i am a very sweet and")
+    EXPECTED_OUTPUTS.add("Hello my name is nathalie, I am a young girl from")
+
+    disable_exllama = True
+    desc_act = True
+
+    def test_generate_quality(self):
+        # act_order don't work with qlinear_cuda kernel
+        pass
+
+    def test_serialization(self):
+        # act_order don't work with qlinear_cuda kernel
+        pass
+
+    def test_exllama_serialization(self):
+        """
+        Test the serialization of the model and the loading of the quantized weights with exllama kernel
+        """
+        from accelerate import init_empty_weights
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            self.quantizer.save(self.quantized_model, tmpdirname)
+            self.quantized_model.config.save_pretrained(tmpdirname)
+            with init_empty_weights():
+                empty_model = AutoModelForCausalLM.from_config(
+                    AutoConfig.from_pretrained(self.model_name), torch_dtype=torch.float16
+                )
+            empty_model.tie_weights()
+            quantized_model_from_saved = load_quantized_model(
+                empty_model, save_folder=tmpdirname, device_map={"": 0}, disable_exllama=False
+            )
+            self.check_inference_correctness(quantized_model_from_saved)
+
+    def test_exllama_max_input_length(self):
+        """
+        Test if the max_input_length works with exllama + act_order
+        """
+        from accelerate import init_empty_weights
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            self.quantizer.save(self.quantized_model, tmpdirname)
+            self.quantized_model.config.save_pretrained(tmpdirname)
+            with init_empty_weights():
+                empty_model = AutoModelForCausalLM.from_config(
+                    AutoConfig.from_pretrained(self.model_name), torch_dtype=torch.float16
+                )
+            empty_model.tie_weights()
+            quantized_model_from_saved = load_quantized_model(
+                empty_model, save_folder=tmpdirname, device_map={"": 0}, disable_exllama=False, max_input_length=4028
+            )
+
+            prompt = "I am in Paris and" * 1000
+            inp = self.tokenizer(prompt, return_tensors="pt").to(0)
+            self.assertTrue(inp["input_ids"].shape[1] > 4028)
+            with self.assertRaises(RuntimeError) as cm:
+                quantized_model_from_saved.generate(**inp, num_beams=1, min_new_tokens=3, max_new_tokens=3)
+                self.assertTrue("temp_state buffer is too small" in str(cm.exception))
```

Comment:
The error shown from AutoGPTQ is irrelevant if the model is loaded through transformers right? Should I do a fix on AutoGPTQ side to have the exllama_set_max_input_length method behave well with transformers?

Reply:
Calling `exllama_set_max_input_length` directly on a transformers model will fail, see: https://github.com/PanQiWei/AutoGPTQ/issues/315

Reply:
Why is this fine? The error shown by AutoGPTQ suggests to use `exllama_set_max_input_length` which does not work with transformers.

Reply:
What is the fix that you want to do ? I think i'm missing something 😅

Reply:
That would be great indeed ! The workaround that is used for now in optimum/transformers is: 
```py 
class StoreAttr(object):
    pass

model.quantize_config = StoreAttr()
model.quantize_config.desc_act = self.desc_act
```